sunpy/physics/limb_darkening.py

```diff
@@ -0,0 +1,87 @@
+import numpy as np
+from astropy import units as u
+from astropy.coordinates import SkyCoord
+
+__author__ = ["Norbert Gyenge"]
+
+
+def limbdark(observation, limb_cut=True):
+    """
+    Calculates the limb darkening function for a specified wavelength
+    and removes any limb darkening effect from the input image.
+    A 5th order polynomial is fit to the limb darkening constants obtained from
+    Astrophysical Quantities by Allan. Only in the wavelength
+    range between 4000 and 15000 A.
+
+    Parameters
+    ----------
+    observation : `sunpy.map.GenericMap`.
+        Input map object
+    limb_cut : `bool`
+        If `True` background (r > 1) values will be repleaced by np.nan.
+
+    Returns
+    -------
+    new_observation : `sunpy.map.GenericMap`
+        A new corrected Map.
+
+    Notes
+    -----
+    IDL code equavalent:
+        http://hesperia.gsfc.nasa.gov/ssw/packages/nrl/idl/nrlgen/analysis/limb_dark.pro
+
+    Reference
+    ---------
+    Cox, Arthur N., Allen's astrophysical quantities, 4th ed. Publisher:
+        New York: AIP Press; Springer, 2000. ISBN: 038798746
+
+    Examples
+    --------
+    >>> import sunpy.map
+    >>> from limb_darkening import limbdark
+    >>> mymap = sunpy.map.Map('test.fits')
+    >>> newmap = limb_darkening(mymap, limb_cut=True)
+    >>> mymap.peek()
+    >>> newmap.peek()
+    """
+
+    if observation.wavelength < 400 * u.nm or observation.wavelength > 1500 * u.nm:
+        raise ValueError("The wavelength of the observation must be between 400 and 1500 nm.")
+
+    wavelength = observation.wavelength.value
```

Comment:
Assuming this is expecting angstroms you should convert units.
```python
wavelength = observation.wavelength.to(u.AA).value
```
to ensure you are getting the right units for the rest of the calculations. This should also be done as good practice for every other use of `.value`.